File: server/src/main/java/com/squat/server/dto/CoachingResponse.java
package com.squat.server.dto;

/// 단일, 장기 분석 공통 응답 DTO
public class CoachingResponse {
    private String coachingType; // "SINGLE" 또는 "AGGREGATE"
    private int totalSessions;   // 분석 대상 세트 수
    private int totalSuccessCount;
    private int totalWaistErrorCount;
    private int totalDepthErrorCount;
    private int totalGoodMorningCount;
    private String coachingMessage;

    public CoachingResponse() {}

    public CoachingResponse(String coachingType, int totalSessions, int totalSuccessCount,
                            int totalWaistErrorCount, int totalDepthErrorCount,
                            int totalGoodMorningCount, String coachingMessage) {
        this.coachingType = coachingType;
        this.totalSessions = totalSessions;
        this.totalSuccessCount = totalSuccessCount;
        this.totalWaistErrorCount = totalWaistErrorCount;
        this.totalDepthErrorCount = totalDepthErrorCount;
        this.totalGoodMorningCount = totalGoodMorningCount;
        this.coachingMessage = coachingMessage;
    }

    // Getter & Setter
    public String getCoachingType() { return coachingType; }
    public int getTotalSessions() { return totalSessions; }
    public int getTotalSuccessCount() { return totalSuccessCount; }
    public int getTotalWaistErrorCount() { return totalWaistErrorCount; }
    public int getTotalDepthErrorCount() { return totalDepthErrorCount; }
    public int getTotalGoodMorningCount() { return totalGoodMorningCount; }
    public String getCoachingMessage() { return coachingMessage; }
}

File: app/lib/dtos/coaching_response.dart
class CoachingResponse {
  final String coachingType;
  final int totalSessions;
  final int totalSuccessCount;
  final int totalWaistErrorCount;
  final int totalDepthErrorCount;
  final int totalGoodMorningCount;
  final String coachingMessage;

  CoachingResponse({
    required this.coachingType,
    required this.totalSessions,
    required this.totalSuccessCount,
    required this.totalWaistErrorCount,
    required this.totalDepthErrorCount,
    required this.totalGoodMorningCount,
    required this.coachingMessage,
  });

  factory CoachingResponse.fromJson(Map<String, dynamic> json) {
    return CoachingResponse(
      coachingType: json['coachingType'] ?? 'SINGLE',
      totalSessions: json['totalSessions'] ?? 1,
      totalSuccessCount: json['totalSuccessCount'] ?? 0,
      totalWaistErrorCount: json['totalWaistErrorCount'] ?? 0,
      totalDepthErrorCount: json['totalDepthErrorCount'] ?? 0,
      totalGoodMorningCount: json['totalGoodMorningCount'] ?? 0,
      coachingMessage: json['coachingMessage'] ?? '코칭 메시지가 없습니다.',
    );
  }
}